### PR TITLE
[CS] - Get cold stake address from script and added CS to Address Indexer

### PR DESCRIPTION
**Pull request by DennisAMenace**
Added ability to get cold stake address from script
Added cold stake to address indexer.

### CODE REVIEW

File: src/NBitcoin/Script.cs [hunk 1]
```diff
@@ -444,7 +444,7 @@ public Script(IEnumerable<byte> data)
 
         public Script(byte[] data, bool compressed)
         {
-            if(!compressed)
```
Comment (dangershony) [inline src/NBitcoin/Script.cs]: this file has lots of unrelated formatting changes, any chance to undo those (formatting better do in its own PR, especially in such a core class)

**Comment by dangershony** [comment]
Perhaps the better approach is to override ScriptAddressReader in the cold staking project and there provide the ability to fetch the hot/cold keys.

File: src/Stratis.Bitcoin/Consensus/ScriptAddressResult.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿namespace Stratis.Bitcoin.Consensus
+{
+    public class ScriptAddressResult
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Consensus/ScriptAddressResult.cs]: How about `ScriptAddressResult` hold only an address, and we make an override of that `ScriptColdStakeAddressResult :  ScriptAddressResult` with the additional addresses, then when calling just check the type.

File: src/NBitcoin/Script.cs (hunk 1 above)
Comment (DennisAMenace) [reply src/NBitcoin/Script.cs]: I will undo this and update the changes, force of habit, I will keep this in mind going forward, thanks.

**Comment by DennisAMenace** [comment]
> Perhaps the better approach is to override ScriptAddressReader in the cold staking project and there provide the ability to fetch the hot/cold keys.

I'm not following you. the cold staking project references NBitcoin so this would need to be in NBitcoin.

File: src/Stratis.Bitcoin/Consensus/ScriptAddressResult.cs (hunk 2 above)
Comment (DennisAMenace) [reply src/Stratis.Bitcoin/Consensus/ScriptAddressResult.cs]: Are you thinking to do this so we don't need to ToString() when calling GetDestinationAddress to when retrieving the address?

**Comment by dangershony** [comment]
@DennisAMenace I think we need to understand this PR better, in theory if we do it right we should not need make changes to the script class, you approach on the ScriptResult is a step in the right direction though.

**Comment by DennisAMenace** [comment]
> @DennisAMenace I think we need to understand this PR better, in theory if we do it right we should not need make changes to the script class, you approach on the ScriptResult is a step in the right direction though.

Sounds good bossman, LMK how to proceed.

**Comment by DennisAMenace** [comment]
Closed this in favor of PR https://github.com/block-core/blockcore/pull/141
